src/BlockTypes/MiniCart.php

```diff
@@ -72,6 +72,7 @@ public function __construct( AssetApi $asset_api, AssetDataRegistry $asset_data_
 	protected function initialize() {
 		parent::initialize();
 		add_action( 'wp_loaded', array( $this, 'register_empty_cart_message_block_pattern' ) );
+		add_action( 'hooked_block_types', array( $this, 'register_auto_insert' ), 10, 4 );
```

Comment:
I don't know how/why this worked but the hook is a filter according to [the docs](https://make.wordpress.org/core/2023/10/15/introducing-block-hooks-for-dynamic-blocks/) and I've only just realised this was added as an action.

Reply:
It's a little known fact that `add_action` is [actually a wrapper around `add_filter`](https://github.com/WordPress/wordpress-develop/blob/e43275b61c6e3a8d5657b6d21eb99bf1bafd52b2/src/wp-includes/plugin.php#L441). To make the intent clear, it would be better to use the appropriate function though.

Reply:
TIL